lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,503 @@
+use anyhow::Context;
+use cargo_metadata::{CargoOpt, MetadataCommand};
+use clap::Parser;
+use std::collections::HashMap;
+use std::path::Path;
+use std::path::PathBuf;
+
+static NOTE: &str =
+    "# See more keys and definitions at https://docs.wasmer.io/ecosystem/wapm/manifest";
+
+const NEWLINE: &str = if cfg!(windows) { "\r\n" } else { "\n" };
+
+/// CLI args for the `wasmer init` command
+#[derive(Debug, Parser)]
+pub struct Init {
+    /// Initialize wapm.toml for a library package
+    #[clap(long, group = "crate-type")]
+    pub lib: bool,
+    /// Initialize wapm.toml for a binary package
+    #[clap(long, group = "crate-type")]
+    pub bin: bool,
+    /// Initialize an empty wapm.toml
+    #[clap(long, group = "crate-type")]
+    pub empty: bool,
```

Comment:
wouldn't it be better to have a 3 state type here, as the init is either for a binary, a lib or empty, but need to be 1 of that and not a combination? Or can it be both for bin & lib?

Reply:
It's not possible with clap: https://github.com/clap-rs/clap/issues/2621

But clap automatically watches that you can't specify both `--bin` and `--lib` for example

Reply:
Ah, with "group" thingy I assume. Nice.